portal-impl/src/com/liferay/portal/security/permission/AdvancedPermissionChecker.java

```diff
@@ -764,6 +764,12 @@ else if ((groupId > 0) &&
 			newIndividualResourcePrimKey = name;
 		}
 
+		else if (primKey.equals(
+					String.valueOf(ResourceConstants.PRIMKEY_DNE))) {
+
+			newIndividualResourcePrimKey = name;
+		}
+
```

Comment:
This condition should belong into the condition below. It's the same scenario 

Seems we check general permissions with primKey `-1`, `0` and `companyId`.

Thanks.